src/main/java/org/springframework/integration/aws/support/S3FileInfo.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.aws.support;
+
+import org.springframework.integration.file.remote.AbstractFileInfo;
+
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+/**
+ * An Amazon S3 {@link org.springframework.integration.file.remote.FileInfo} implementation.
+ * @author Christian Tzolov
+ */
+public class S3FileInfo extends AbstractFileInfo<S3ObjectSummary> {
+
+	private S3ObjectSummary s3ObjectSummary;
+
+	public S3FileInfo(S3ObjectSummary s3ObjectSummary) {
+		this.s3ObjectSummary = s3ObjectSummary;
+	}
+
+	@Override
+	public boolean isDirectory() {
+		return (s3ObjectSummary != null ? s3ObjectSummary.getKey().endsWith("/") : false);
```

Comment:
Can be simplified.
And I'm not sure that `/` is panacea to determine the dir...

The S3 has one dir - `bucket`. Everything rest is keys and when we scan the `bucket` for files we receive the entire path, even if it is artificial sub-dir:

```
bucket/foo.txt
bucket/bar/foo.txt
bucket/bar/baz/foo.txt
```
will be listed only once, without any recursion. So, we will never receive something like `bucket/bar/`

Therefore this method should always return `false`.

I won't mind if I miss something.
Thanks